### DISCUSSION TITLE
How to use local presets?

### DISCUSSION THREAD

**Discussion by marinalimeira** [renovatebot/renovate#13408] [Request Help]
I am trying to share a [preset file](https://docs.renovatebot.com/config-presets/) for different environments in the same repo.

I have the following structure:

```
.
├── config-dev.js
├── config-prod.js
└── default.json
```

At `config-prod.js`, it imports `github>organization/repo`, but I'd like to use `default.json` locally in the `config-dev.json`.  My current solution is to import it as `github>organization/repo#branch`. It's not ideal for local development as I need to `git push` at every minor change I want to test, it's error-prone and it becomes a pain for the team in these last weeks...

Am I missing something in the docs or isn't possible to use a `default.json` file locally? If so, how should I add it to the `extends`?

**Comment by rarkins** [comment]
Renovate's definition of local means "same GitHub/gitlab host" and not local file system or local repo.

Can you describe in more detail what you are trying to do and how you are testing? Eg seems like you're using presets for global config and not repo config?

**Comment by marinalimeira** [comment]
> Renovate's definition of local means "same GitHub/gitlab host" and not local file system or local repo.

Thank you for clarifying!

> Can you describe in more detail what you are trying to do and how you are testing? Eg seems like you're using presets for global config and not repo config?

Yes, I am not adding any Renovate configuration on the repositories, I have all the configuration in one repo, and it has two workflows now:

1. I run nightly the configurations in `config-prod.js`, running in some repos.
```
module.exports = {
  extends: [
    "config:base", "github>org/repo"
  ],

  // Do not require a config file
  requireConfig: false,
  // Disable Onboarding PR
  onboarding: false,

  repositories: []
};
```
  
2. I run in my local machine the configuration from `config-dev.js`, in some cloned repositories, while experimenting with Renovate.

Both share the same `default.json`, I don't wanna mess with prod - the "real" - repos while experimenting. I also use `ln -s` to use the file from the repository with the configuration, in renovate, like this:

```
> renovate/
ln -S ../repo/config-dev.js config.js

> renovate/
ln -S ../repo/default.json default.json
```

Does this workflow even make sense? 😅  I am trying to find an optimal way given there are other devs experimenting with this.

**Reply by rarkins** [reply]
So your "real" requirement is that you can test out new config without needing to do commits to repositories? If so, why can't you edit the *.js files locally temporarily, then commit the same config into the repositories once you've tested it?
